### PR TITLE
Add ManagedClusterSetBinding API

### PR DESCRIPTION
**Pull request by elgnay**

### CODE REVIEW

File: cluster/v1alpha1/types.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:scope="Namespaced"
+
+// ManagedClusterSetBinding projects a managed cluster set into a certain namespace.
```
Comment (pmorie) [inline cluster/v1alpha1/types.go]: Good start at the description, I do think that we should enrich this with information about how users can expect to use these resources. Do they act the same as ManagedClusters? What do they allow you to do? I know we're still sorting these properties out, but let's record the key ones here (in subsequent PRs) as we do.

File: cluster/v1alpha1/types.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the attributes of ManagedClusterSetBinding.
+	Spec ManagedClusterSetBindingSpec `json:"spec"`
+}
+
+// ManagedClusterSetBindingSpec defines the attributes of ManagedClusterSetBinding.
+type ManagedClusterSetBindingSpec struct {
+	// ClusterSet is the name of the ManagedClusterSet to bind.
```
Comment (deads2k) [inline cluster/v1alpha1/types.go]: you need more doc around what dangling references do, whether there are any native restrictions against the clusterset name (can anyone bind any clusterset).  Whether instance names are required to match the clusterset name.

**Comment by deads2k** [comment]
/approve

The doc needs improvement.  I'll leaving tagging lgtm with @qiujian16 

/assign @qiujian16

File: cluster/v1alpha1/types.go (hunk 2 above)
Comment (elgnay) [reply cluster/v1alpha1/types.go]: Doc added

File: cluster/v1alpha1/types.go [hunk 3]
```diff
@@ -99,3 +99,44 @@ type ManagedClusterSetList struct {
...
+
+	// Spec defines the attributes of ManagedClusterSetBinding.
+	Spec ManagedClusterSetBindingSpec `json:"spec"`
+}
+
+// ManagedClusterSetBindingSpec defines the attributes of ManagedClusterSetBinding.
+type ManagedClusterSetBindingSpec struct {
+	// ClusterSet is the name of the ManagedClusterSet to bind. The instance name of the
+	// ManagedClusterSetBinding doesn't have to match the target ClusterSet. User is
+	// allowed to bind a ManagedClusterSet if they have an RBAC rule to GET on the
```
Comment (qiujian16) [inline cluster/v1alpha1/types.go]: I would say like ```User is allowed to set or update this field if they have an RBAC rule...```

Comment (elgnay) [reply cluster/v1alpha1/types.go]: Updated

**Pull request merged**